connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -142,6 +155,57 @@ public WorkerSourceTask(ConnectorTaskId id,
         this.sourceTaskMetricsGroup = new SourceTaskMetricsGroup(id, connectMetrics);
         this.producerSendException = new AtomicReference<>();
         this.isTopicTrackingEnabled = workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG);
+        this.topicCreation = TopicCreation.newTopicCreation(workerConfig, topicGroups);
+    }
+
+    public static class TopicCreation {
+        private static final TopicCreation EMPTY =
+                new TopicCreation(false, null, Collections.emptyMap(), Collections.emptySet());
+
+        private final boolean isTopicCreationEnabled;
+        private final NewTopicCreationGroup defaultTopicGroup;
+        private final Map<String, NewTopicCreationGroup> topicGroups;
+        private final Set<String> topicCache;
+
+        protected TopicCreation(boolean isTopicCreationEnabled,
+                                NewTopicCreationGroup defaultTopicGroup,
+                                Map<String, NewTopicCreationGroup> topicGroups,
+                                Set<String> topicCache) {
+            this.isTopicCreationEnabled = isTopicCreationEnabled;
+            this.defaultTopicGroup = defaultTopicGroup;
+            this.topicGroups = topicGroups;
+            this.topicCache = topicCache;
+        }
+
+        public static TopicCreation newTopicCreation(WorkerConfig workerConfig,
+                                                     Map<String, NewTopicCreationGroup> topicGroups) {
+            if (!workerConfig.topicCreationEnable() || topicGroups == null) {
+                return EMPTY;
+            }
+            Map<String, NewTopicCreationGroup> groups = new LinkedHashMap<>(topicGroups);
+            groups.remove(DEFAULT_TOPIC_CREATION_GROUP);
+            return new TopicCreation(true, topicGroups.get(DEFAULT_TOPIC_CREATION_GROUP), groups, new HashSet<>());
+        }
+
+        public static TopicCreation empty() {
+            return EMPTY;
+        }
```

Comment:
Nit: Since this is static, how about moving to before the member fields and methods?